File: 1512 - Number of Good Pairs/Number of Good Pairs.cpp
class Solution {
public:
    int numIdenticalPairs(const vector<int>& nums) {
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); ++i) {
            ++m[nums[i]];
        }
        int result = 0;
        for (const auto p : m) {
            result += (p.second - 1) * (p.second) / 2;
        }
        return result;
    }
};

File: 1512 - Number of Good Pairs/Number of Good Pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        d = {}
        for el in nums:
            d[el] = d.get(el, 0) + 1
        result = 0
        for value in d.values():
            result += value * (value - 1) // 2
        return result

